frontend/src/App.js

```diff
@@ -49,7 +52,25 @@ FROM ( SELECT MONTH(committer_when) as month,
   }
 
   setResult(key, result) {
-    if (!this.state.results.has(key)) {
+    const { results, history } = this.state;
+    const historyIdx = history.findIndex(i => i.key === key);
```

Comment:
I think it might be cleaner (and faster) if we make `state.history` a Map, as `results`. What do you think?

Reply:
it's not easy to serialize/deserialize map. In next PR I changed results too.